core/src/main/java/discord4j/core/object/entity/channel/AllowedMentions.java

```diff
@@ -0,0 +1,222 @@
+    private final Possible<List<Snowflake>> userIds;
+    private final Possible<List<Snowflake>> roleIds;
+
+    private AllowedMentions(final Possible<List<AllowedMentions.Type>> parse, final Possible<List<Snowflake>> userIds,
+                            final Possible<List<Snowflake>> roleIds) {
+        this.parse = parse;
+        this.userIds = userIds;
+        this.roleIds = roleIds;
+    }
+
+    private <T, U> List<T> mapList(final List<U> list, final Function<? super U, ? extends T> mapper) {
+        final List<T> data = new ArrayList<>(list.size());
+        list.forEach(u -> data.add(mapper.apply(u)));
+        return data;
+    }
+
+    /**
+     * Maps this {@link AllowedMentions} object to a {@link AllowedMentionsData} JSON
+     * @return JSON object
+     */
+    public AllowedMentionsData toData() {
+        final ImmutableAllowedMentionsData.Builder builder = AllowedMentionsData.builder();
+        if (!parse.isAbsent()) {
+            builder.parse(mapList(parse.get(), Type::getRaw));
+        }
+        if (!userIds.isAbsent()) {
+            builder.users(mapList(userIds.get(), Snowflake::asString));
+        }
+        if (!roleIds.isAbsent()) {
+            builder.roles(mapList(roleIds.get(), Snowflake::asString));
+        }
+        return builder.build();
+    }
+
+    public static class Builder {
+
+        private Possible<List<AllowedMentions.Type>> parse = DEFAULT_TYPES;
+        private Possible<List<Snowflake>> userIds = DEFAULT_USERS;
+        private Possible<List<Snowflake>> roleIds = DEFAULT_ROLES;
+
+        /**
+         * Add a type to the parsed types list
+         * @param type the type to parse
+         * @return this builder
+         */
+        public Builder parseType(final AllowedMentions.Type type) {
+            if (parse.isAbsent()) {
+                parse = Possible.of(new ArrayList<>());
+            }
+            parse.get().add(type);
+            return this;
+        }
+
+        /**
+         * Add a user to the allowed users list
+         * @param userId the user to allow
+         * @return this builder
+         */
+        public Builder allowUser(final Snowflake userId) {
+            if (userIds.isAbsent()) {
+                userIds = Possible.of(new ArrayList<>());
+            }
+            userIds.get().add(userId);
+            return this;
+        }
+
+        /**
+         * Add a role to the allowed roles list
+         * @param roleId the role to allow
+         * @return this builder
+         */
+        public Builder allowRole(final Snowflake roleId) {
+            if (roleIds.isAbsent()) {
+                roleIds = Possible.of(new ArrayList<>());
+            }
+            roleIds.get().add(roleId);
+            return this;
+        }
+
+        /**
+         * Add types to the parsed types list
+         * @param type the types to parse
+         * @return this builder
+         */
+        public Builder parseType(final AllowedMentions.Type... type) {
```

Comment:
I think it'd be better for these varargs methods to instead accept collections (either List or Set depending on the result of the other review comment).

Reply:
I don't think we need to have collections for that, as there are just 3 types, which putting into an collection to addAll to another collection would be overhead in the usage.
For users and roles it could be any number of entries, but in the usual case you're building this from a context without a collection. So it might be more convenient to have varargs.

If you insist of having a collection, I'm fine with that and I'll implement it like that. Just my thoughts above.
Let me know how you think about this @darichey